### PR TITLE
Add raw VR consts

### PR DESCRIPTION
**Pull request by bpeake-illuscio**
Hello! I love this library, I hope you do not mind me opening a few PR's for some quick quality of life features. I'll start with this one and once I get some feedback move on to a few others.

In using this library, I've found it useful to have a set of aliases for all the raw value representations defined in the dicom spec,
for instance:

```
if element.RawValueRepresentation == vrraw.PersonName {
    // do something
}
```

This PR adds a vrraw package with a set of string consts for inspecting the RawValueRepresentation without having to worry about typos.

The package name "vrraw" was chosen over "vr" to:

1. Match more closely element.RawValueRepresentation
2. Avoid importing vr into the namespace and removing that value from variable name availability

I also considered the possibility of making a VRRaw custom type off of string, like VRKind, and have these consts use that type. In such a case, we would also want to change element.RawValueRepresentation to the new type, but such a change could conceivably break code.

This most common case would still work:

```
// Go will cast the literal to the correct type at compile-time.
if element.RawValueRepresentation == "PN" {
    // do something
}
```

But this case would break:

```
func ReturnsPN() string {
    return "PN"
}

// You can't compare values of two different types, and non-literals are not automatically casted. So this would break.
if element.RawValueRepresentation == ReturnsPN() {
    // do something
}
```

As would this:

```
func TakesVR(vr string) string {
    return "PN"
}

// This will break: passed variables cannot be implicitly casted.
TakesVR(element.RawValueRepresentation)
```

Therefore, in order to keep backwards compatibility I kept this PR small in scope: make string consts for the Dicom-spec-defined Value Representations, but do not make these values their own type.

Let me know what you think! Thank you!

### CODE REVIEW

**Comment by bpeake-illuscio** [comment]
Just noticed this seems related to #152. I would be happy to refactor this is a proper enum, to more closely align with the open ticket.

I had been worried about opening a PR with a breaking change, but if it is something you had planned already, just let me know and I am happy to revise it.

File: pkg/vrraw/vrConsts.go
Comment (suyashkumar) [inline pkg/vrraw/vrConsts.go:1]: nit: could you rename this file to just be `vrraw.go`? For small packages, typically I see the convention that the primary go source file contains the same name as the package. Also, I tend to follow the same naming conventions for packages for file names--that is, avoid camelcase (and typically stick to lower case):

https://golang.org/doc/effective_go.html#names

File: pkg/vrraw/vrConsts.go
Comment (suyashkumar) [inline pkg/vrraw/vrConsts.go:38]: I'm not so sure about this from a stylistic perspective--it might be nice to have one choice per VR, so client code isn't bouncing back and forth between two symbols that are actually the same under the hood (but may not appear that way when reading the code). What do you think?

File: pkg/vrraw/vrConsts_test.go
Comment (suyashkumar) [inline pkg/vrraw/vrConsts_test.go:15]: I would recommend using anonymous structs for tests cases, as this is a common convention throughout Go code (in the stdlib too). It also reduces boilerplate code a bit.

See https://github.com/golang/go/wiki/TableDrivenTests#example-of-a-table-driven-test

File: pkg/vrraw/vrConsts_test.go
Comment (suyashkumar) [inline pkg/vrraw/vrConsts_test.go:180]: also as seen in https://github.com/golang/go/wiki/TableDrivenTests#example-of-a-table-driven-test, I would recommend structuring this as

```go
for _, tc := range testCases {
    t.Run(tc.Name, t.Run(t *testing.T) {
        // The test body here
    })
}
```

File: pkg/vrraw/vrConsts_test.go
Comment (suyashkumar) [inline pkg/vrraw/vrConsts_test.go:1]: You know, I'm not sure if this needs to be tested in this way--this is just a definition of constants. The test isn't testing any functionality, it's just testing that constant values are set to...constant values. I would consider just dropping the test for this.

**Review by suyashkumar — COMMENTED** [review]
Thanks for the PR and the kind words about the library! I think it's reasonable to hold onto these constants in a `vrraw` package, and left some comments below. Ultimately I think what really needs to happen here is what I mentioned in #152 as you saw yourself. Either way though, holding onto these constants is nice and useful imo!

File: pkg/vrraw/vrConsts.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrConsts.go:1]: No probelm. Will change.

File: pkg/vrraw/vrConsts.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrConsts.go:38]: I'll keep the verbose ones, and leave comments about the other names. My original thought was for people new to the DICOM spec, URL / URI might be more familiar, but that's a good point about redundant symbols.

File: pkg/vrraw/vrConsts_test.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrConsts_test.go:15]: No problem, I think I tend to lean away from anonymous structs as I prefer the self-documenting / explicit nature of giving structs a name, but that's just a stylistic preference on my part and a little out of line with how a lot of Go is written. I'll switch out my table tests on other PRs to use anonymous structs as well.

File: pkg/vrraw/vrConsts_test.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrConsts_test.go:180]: Will do.

File: pkg/vrraw/vrConsts_test.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrConsts_test.go:1]: I agree that looking at it after the fact, it seems a bit silly, but my thought process when testing it was that if I forced myself to re-type the expected values for a unit test, I would be more likely to catch typos in the actual values.

As it so happened, I did fat finger "UV" as "UC" or something, so the act of writing the test did catch that.

I would lean toward keeping it, if anyone ever goes in and makes changes, this becomes codegen in some way, or an IDE accidentally alters the string values of a const during a refactor of a local variable somewhere (I've had this happen), the unit tests will be useful to have.

That being said I am also happy to remove it.

File: pkg/vrraw/vrConsts_test.go
Comment (suyashkumar) [reply pkg/vrraw/vrConsts_test.go]: Thanks! Yeah, my feeling is that in this case, it's typically better to stick with idiomatic code patterns found throughout the Go ecosystem (e.g. in [stdlib](https://github.com/golang/go/wiki/TableDrivenTests#example-of-a-table-driven-test) and other common Go libraries)--this means the code tends to be more "familiar" at first glance to folks working in Go (and this pattern is quite common in my experience--and is common throughout this library).

As for the readability of unnamed structs, I think the thinking is that the struct is only relevant to that particular test scope, so it may not be that important what the name is (similar to the reasoning behind Go's short variable names for variables used in a small well defined scope: https://github.com/golang/go/wiki/CodeReviewComments#variable-names

File: pkg/vrraw/vrraw.go
Comment (suyashkumar) [inline pkg/vrraw/vrraw.go:41]: nit: extra space
```suggestion
	// UniversalResourceLocator, URI and URL.
```

File: pkg/vrraw/vrraw.go
Comment (suyashkumar) [inline pkg/vrraw/vrraw.go:9]: A thought occurred to me: these are all exported constants, therefore should have a comment describing the symbol.

As discussed in [Effective Go](https://golang.org/doc/effective_go.html#commentary):
 > Every exported (capitalized) name in a program should have a doc comment.

It might seem a bit much to do this, but I am actually thinking it could be nice documentation to have here and may save some folks a google search or trip to the [standard](http://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_6.2) (though they'll probably be doing that anyway haha).

If you have the time, maybe we can include a short one liner description for these (based on the standard description) and include a link to the relevant portion of the standard: http://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_6.2

I can also follow up and do this in another PR (I won't block the PR on this--just thought I'd mention).

File: pkg/vrraw/vrraw.go
Comment (suyashkumar) [inline pkg/vrraw/vrraw.go:37]: Comments should start with the exported name first, e.g.

```go
// UniqueIdentifier ...
```

File: pkg/vrraw/vrraw_test.go
Comment (suyashkumar) [inline pkg/vrraw/vrraw_test.go:155]: nit: you can consider a shorter variable name for `thisCase` if you prefer (see https://github.com/golang/go/wiki/CodeReviewComments#variable-names), since it is used in a small scope within a for loop (commonly you'll see `tc` here as an abbreviation for "test case" or "this case").

This sort of thing is definitely one of the quirks of Go that it takes getting used to seeing, but I've found in practice it doesn't hurt readability when used in the right scopes. 

```suggestion
	for _, tc := range testCases {
		t.Run(tc.Expected, func(t *testing.T) {
			if tc.Expected != tc.VR {
				t.Errorf("expected %v, got %v", tc.Expected, tc.VR)
```

File: pkg/vrraw/vrConsts_test.go
Comment (suyashkumar) [reply pkg/vrraw/vrConsts_test.go]: Your point is well-taken, though I'm still a bit on the fence about it. This is because the test doesn't really enforce any _functionality_, but merely is a second definition of the constants and enforce that both change together. Since these constants are meaningful, I could see a test where we maybe check that they are consistent with another implementation or some external reference or something, but an internal reference seems a bit odd to me. 

Setting a constant to an incorrect literal value seems like something that needs to be checked in a Code Review regardless of if the test exists or not. 

That being said, it doesn't really hurt, other than some additional overhead whenever a new VR is added. Since that's unlikely to be a common occurence, perhaps leaving the test is okay. Also, if we decide it's not needed, we can always easily remove it in a future refactor. 

I'd lean towards not adding it given the above, but in this case, I'll leave the final decision up to you and won't block the submission on it. It's a bit odd imo, but not actively harmful.

**Review by suyashkumar — COMMENTED** [review]
Thanks! This is generally looking good to me. Apologies for the back and forth. Had a couple more recommendations for your consideration. After resolving those we should be good to submit!

File: pkg/vrraw/vrraw.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrraw.go:41]: Good catch!

File: pkg/vrraw/vrraw.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrraw.go:9]: I don't think it seems a bit much at all, and is an excellent idea. Will add and push.

File: pkg/vrraw/vrraw.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrraw.go:37]: Copy

File: pkg/vrraw/vrraw_test.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrraw_test.go:155]: No problem.

**Comment by bpeake-illuscio** [comment]
No apologies necessary on the back and forth, as a user of the lib, it makes me feel good you are thorough on your code reviews! Will be pushing changes shortly.

**Comment by bpeake-illuscio** [comment]
For your note on the Tests (git hub doesn't have a reply field for it, strangely): I will remove them in this update

File: pkg/vrraw/vrConsts_test.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrConsts_test.go]: I will remove them in this update.

File: pkg/vrraw/vrraw_test.go
Comment (bpeake-illuscio) [reply pkg/vrraw/vrraw_test.go]: Since the test were removed, no need to fix, but I will update my other PRs.

**Comment by bpeake-illuscio** [comment]
Changes are up!

**Review by suyashkumar — APPROVED** [review]
Looks good to me, thank you for the fantastic comments!

**Comment by suyashkumar** [comment]
Merged! 🎉

**Comment by suyashkumar** [comment]
Updating usages in the main library as well (#172).

**Pull request merged**
